internal/dag/status_test.go

```diff
@@ -2441,3 +2448,417 @@ func TestDAGStatus(t *testing.T) {
 		},
 	})
 }
+
+func TestGatewayAPIDAGStatus(t *testing.T) {
+
+	type testcase struct {
+		objs []interface{}
+		want []metav1.Condition
+	}
+
+	run := func(t *testing.T, desc string, tc testcase) {
+		t.Helper()
+		t.Run(desc, func(t *testing.T) {
+			t.Helper()
+			builder := Builder{
+				Source: KubernetesCache{
+					RootNamespaces: []string{"roots", "marketing"},
+					FieldLogger:    fixture.NewTestLogger(t),
+					ConfiguredGateway: types.NamespacedName{
+						Namespace: "contour",
+						Name:      "projectcontour",
+					},
+					gateway: &gatewayapi_v1alpha1.Gateway{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "contour",
+							Namespace: "projectcontour",
+						},
+						Spec: gatewayapi_v1alpha1.GatewaySpec{
+							Listeners: []gatewayapi_v1alpha1.Listener{{
+								Port:     80,
+								Protocol: "HTTP",
+							}},
+						},
+					},
+				},
+				Processors: []Processor{
+					&IngressProcessor{
+						FieldLogger: fixture.NewTestLogger(t),
+					},
+					&HTTPProxyProcessor{},
+					&GatewayAPIProcessor{
+						FieldLogger: fixture.NewTestLogger(t),
+					},
+					&ListenerProcessor{},
+				},
+			}
+			for _, o := range tc.objs {
+				builder.Source.Insert(o)
+			}
+			dag := builder.Build()
+			t.Logf("%#v\n", dag.StatusCache)
```

Comment:
nit: Do we want to remove this or keep it?

Reply:
I vote remove.

Reply:
removed.